src/algorithms/rhythm/rhythmtransform.cpp

```diff
@@ -49,63 +49,83 @@ void RhythmTransform::compute() {
   vector<vector<Real> >& output = _rhythmTransform.get();
 
   int nFrames = bands.size();
-  int nBands = bands[0].size();
+  // 
+  // Check first that we have a valid populated input to avoid  core dump
+  if (nFrames== 0) {
+    throw EssentiaException("File looks like a completely silent file");
+    return;
+  }
 
-  // Check first that we have a valid populated input
-  // Otherwise core dump could happen.
-  if ((nFrames!=0) && (nFrames!=0))
+  // Gather individual band lengths
+  vector<Real> bandSizes(nFrames);
+  for (int nband=0; nband<nFrames; ++nband) {
+    bandSizes[nband] = bands[nband].size();
+  }
+
+  // Check if a band is empty
+  if (std::find(bandSizes.begin(), bandSizes.end(), 0) != bandSizes.end())  
   {
-    // derive and transpose
-    vector<vector<Real> > bandsDerivative(nBands);
-    for (int band=0; band<nBands; ++band) {
-      vector<Real> derivTemp(nFrames);
-      derivTemp[0] = 0.0;
-      for (int frame=1; frame<nFrames; frame++) {
-        derivTemp[frame] = bands[frame][band]-bands[frame-1][band];
-      }
-      bandsDerivative[band] = derivTemp;
+    std::cout << "A Violation found empty band"<< std::endl;
+    throw EssentiaException("File looks like an empty band");
+    return;
+  }
+
+  // Check for an inconsistent input vector with inner vectors of different length
+  if(!( std::equal(bandSizes.begin() + 1, bandSizes.end(), bandSizes.begin()) ))
+  {
+    std::cout << "A Violation found different length" << std::endl;
```

Comment:
Remove debug message

Reply:
This was already removed (not outdated label)